### PR TITLE
Rename packages and VirtualEnvironment

### PR DESCRIPTION
**Pull request by manuel-jasso**
- Renamed packages and folders:
  - `@locker/near-membrane` => `@locker/near-membrane-base`
  - `@locker/dom-membrane` => `@locker/near-membrane-dom`
  - `@locker/node-membrane` => `@locker/near-membrane-node`

- Renamed `SecureEnvironment` to `VirtualEnvironment` to align with the Realms conversation.

- Require node >= 14.16

### CODE REVIEW

**Review by jdalton — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "typescript": "^4.1.2"
   },
   "engines": {
-    "node": ">=12"
+    "node": ">=14.16"
```
Comment (jdalton) [inline package.json:63]: Is there something specifically added for the `x.16` or can `>=14` work?

Comment (manuel-jasso) [reply package.json:63]: No, nothing special, just being more explicit as of "today", but I haven't tested with 14.1 for instance.

File: package.json [hunk 2]
```diff
@@ -60,22 +60,22 @@
...
   "engines": {
-    "node": ">=12"
+    "node": ">=14"
   },
   "bundlesize": [
     {
-      "path": "packages/near-membrane/lib/near-membrane.min.js",
-      "maxSize": "10 kB",
+      "path": "packages/near-membrane-base/lib/index.min.js",
+      "maxSize": "13 kB",
```
Comment (caridy) [inline package.json:68]: why did the size of the minified increased?

**Review by caridy — APPROVED** [review]
LGTM

**Pull request merged**